diff --git a/src/json-io.c b/src/json-io.c
--- a/src/json-io.c
+++ b/src/json-io.c
@@ -1595,11 +1595,12 @@ parse_file(const char *file, const char *what)
 	goto res_out;
     }
 
-    fd = open(file, O_CLOEXEC | O_RDONLY);
+    fd = open(file, O_RDONLY);
     if (fd == -1) {
 	lmap_err("failed to open '%s': %s", file, strerror(errno));
 	goto res_out;
     }
+    (void) fcntl(fd, F_SETFD, FD_CLOEXEC);
 
     json_tokener_set_flags(jtk, JSON_TOKENER_STRICT);
     do {
